File: sdk/src/main/java/com/cnblogs/sdk/internal/http/annotation/HTML.java
package com.cnblogs.sdk.internal.http.annotation;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

import com.cnblogs.sdk.internal.parser.HtmlParser;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

/**
 * Html类型请求
 * @author RAE
 * @date 2022/02/15
 * Copyright (c) https://github.com/raedev All rights reserved.
 */
@Documented
@Target(METHOD)
@Retention(RUNTIME)
public @interface HTML {

    /**
     * 定义解析器
     */
    Class<? extends HtmlParser<?>> value();
}

File: sdk/src/main/kotlin/com/cnblogs/sdk/internal/annotation/HTML.kt
package com.cnblogs.sdk.internal.annotation

import kotlin.reflect.KClass

/**
 * HTML解析器注解，用于标注接口采用哪个解析器进行解析
 * @author RAE
 * @date 2022/10/04
 * @copyright Copyright (c) https://github.com/raedev All rights reserved.
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
annotation class HTML(
    val value: KClass<*>
)
